pkg/apis/metrics/types.go

```diff
@@ -16,14 +16,427 @@ limitations under the License.
 
-// Placeholder top-level node resource metrics.
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+)
+
+// RawNode holds node-level unprocessed sample metrics.
 type RawNode struct {
 	unversioned.TypeMeta `json:",inline"`
+	// Standard list metadata, applying to the lists of stats.
+	// More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#types-kinds
+	unversioned.ListMeta `json:",inline"`
+	// Reference to the measured Node.
+	NodeRef api.ObjectReference `json:"nodeRef,omitempty"`
+	// Overall machine metrics.
+	Machine RawContainer `json:"machine,omitempty"`
+	// Metrics of system components.
+	SystemContainers []RawContainer `json:"systemContainers,omitempty"`
 }
 
-// Placeholder top-level pod resource metrics.
+// RawPod holds pod-level unprocessed sample metrics.
 type RawPod struct {
 	unversioned.TypeMeta `json:",inline"`
+	// Standard list metadata, applying to the lists of stats.
+	// More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#types-kinds
+	unversioned.ListMeta `json:",inline"`
+	// Reference to the measured Pod.
+	PodRef api.ObjectReference `json:"podRef,omitempty"`
+	// Metrics of containers in the measured pod.
+	Containers []RawContainer `json:"containers,omitempty"`
+}
+
+// RawContainer holds container-level unprocessed sample metrics.
+type RawContainer struct {
+	// Reference to the measured container.
+	Name string `json:"name,omitempty"`
+	// Describes the container resources.
+	Spec ContainerSpec `json:"spec,omitempty"`
+	// Historical metric samples gathered from the container.
+	Stats []ContainerStats `json:"stats,omitempty"`
+}
+
+type ContainerSpec struct {
+	// Time at which the container was created.
+	CreationTime time.Time `json:"creationTime,omitempty"`
+
+	// Other names by which the container is known within a certain namespace.
+	// This is unique within that namespace.
+	Aliases []string `json:"aliases,omitempty"`
+
+	// Namespace under which the aliases of a container are unique (not a Kubernetes namespace).
+	// An example of a namespace is "docker" for Docker containers.
+	Namespace string `json:"namespace,omitempty"`
+
+	// Metadata labels associated with this container.
+	Labels map[string]string `json:"labels,omitempty"`
+
+	HasCpu bool    `json:"hasCpu,omitempty"`
+	Cpu    CpuSpec `json:"cpu,omitempty"`
+
+	HasMemory bool       `json:"hasMemory,omitempty"`
+	Memory    MemorySpec `json:"memory,omitempty"`
+
+	HasCustomMetrics bool         `json:"hasCustomMetrics,omitempty"`
+	CustomMetrics    []MetricSpec `json:"customMetrics,omitempty"`
+
+	// Following resources have no associated spec, but are being isolated.
+	HasNetwork    bool `json:"hasNetwork,omitempty"`
+	HasFilesystem bool `json:"hasFilesystem,omitempty"`
+	HasDiskIo     bool `json:"hasDiskIo,omitempty"`
+
+	// Image name used for this container.
+	Image string `json:"image,omitempty"`
+}
+
+type CpuSpec struct {
+	// Requested cpu shares. Default is 1024.
+	// Units: milli-cpus.
```

Comment:
@vishh Do you know the answer to this? It doesn't even look like this struct is ever used?